trade-finance-manager-api/src/v1/helpers/amendment.helpers.js

```diff
@@ -310,24 +312,49 @@ const firstTaskEmailConfirmation = async (facilityId, amendmentId, auditDetails)
   await api.updateFacilityAmendment(facilityId, amendmentId, payload, auditDetails);
 };
 
-// sends email for first amendment task when pim submit amendment
-const sendFirstTaskEmail = async (taskVariables, auditDetails) => {
-  const { amendment, dealSnapshot, facilityId, amendmentId } = taskVariables;
-  const { tasks } = amendment;
-  const { dealId, exporter } = dealSnapshot;
-
-  // dealId in snapshot for gef and details for bss
-  const ukefDealId = dealSnapshot.ukefDealId || dealSnapshot.details.ukefDealId;
-
-  const firstTask = getFirstTask(tasks);
-  const urlOrigin = process.env.TFM_UI_URL;
-  const templateId = EMAIL_TEMPLATE_IDS.TASK_READY_TO_START;
-
+/**
+ * Sends an email notification for the first task of an amendment.
+ *
+ * This function performs the following operations:
+ * 1. Extracts necessary details from the deal and amendment.
+ * 2. Retrieves the first task from the amendment tasks.
+ * 3. Constructs the email variables and sends the email using the TFM email service.
+ * 4. Updates the flag to indicate that the email has been sent if the email is successfully sent.
+ *
+ * @param {Object} deal - The deal object containing deal and amendment details.
+ * @param {Object} deal.amendment - The amendment details.
+ * @param {Object} deal.dealSnapshot - The snapshot of the deal details.
+ * @param {string} deal.facilityId - The ID of the facility.
+ * @param {string} deal.amendmentId - The ID of the amendment.
+ * @param {Object} deal.dealSnapshot.exporter - The exporter details.
+ * @param {string} deal.dealSnapshot.exporter.companyName - The name of the exporter company.
+ * @param {string} [deal.dealSnapshot.ukefDealId] - The UKEF deal ID for GEF deals.
+ * @param {Object} [deal.dealSnapshot.details] - The details object for BSS deals.
+ * @param {string} [deal.dealSnapshot.details.ukefDealId] - The UKEF deal ID for BSS deals.
+ * @param {Object} auditDetails - The audit details for logging purposes.
```

Comment:
It looks like the params could be simpler perhaps:

```ts
* @param {import('@ukef/dtfs2-common').Deal}
* @param {import("@ukef/dtfs2-common").AuditDetails}
```

Reply:
This might kick up a stink with `Deal` as it may need bulking out to contain these items (but I'm not sure how much JSDocs cares about these things in reality -- so worth trying!

Reply:
Need to get this PR asap so can look into this after merge.

Reply:
This a good shout @ttbarnes unfortunately not all of these properties exist so might be prudent to leave as it, however I have updated for `AuditDetails`.

Reply:
https://github.com/UK-Export-Finance/dtfs2/pull/3899/files